src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -0,0 +1,119 @@
+ * test-level lifecycle methods (i.e., {@code @BeforeEach} and {@code @AfterEach}
+ * methods).
+ *
+ * <h3>Applicability</h3>
+ *
+ * <p>In this context, the term "test method" refers to any method annotated with
+ * {@code @Test}, {@code @RepeatedTest}, {@code @ParameterizedTest},
+ * {@code @TestFactory}, or {@code @TestTemplate}. This extension does not inject
+ * {@code SoftAssertions} or {@code BDDSoftAssertions} arguments into test
+ * constructors or class-level lifecycle methods.
+ *
+ * <h3>Scope</h3>
+ *
+ * <p>The scope of the {@code SoftAssertions} or {@code BDDSoftAssertions} instance
+ * managed by this extension begins when a parameter of type {@code SoftAssertions}
+ * or {@code BDDSoftAssertions} is resolved for a {@code @BeforeEach} method,
+ * test method, or {@code @AfterEach} method. The scope of the instance ends after
+ * all {@code @AfterEach} methods have been executed for the corresponding test.
+ * When the scope ends, {@code assertAll()} will be invoked on the instance to
+ * verify that no soft assertions failed.
+ *
+ * <h3>Example with {@code SoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(SoftAssertions softly) {
+ *       softly.assertThat(2 * 3).isEqualTo(0);
+ *       softly.assertThat(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.assertThat(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * <h3>Example with {@code BDDSoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(BDDSoftAssertions softly) {
+ *       softly.then(2 * 3).isEqualTo(0);
+ *       softly.then(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.then(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * @author Sam Brannen
+ * @since 3.13
+ */
+public class SoftAssertionsExtension implements ParameterResolver, AfterEachCallback {
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		boolean isTestLevel = extensionContext.getTestMethod().isPresent();
+		boolean isSoftAssertionsParameter = parameterContext.getParameter().getType() == SoftAssertions.class;
+		boolean isBddSoftAssertionsParameter = parameterContext.getParameter().getType() == BDDSoftAssertions.class;
+		return (isTestLevel && (isSoftAssertionsParameter || isBddSoftAssertionsParameter));
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		if (parameterContext.getParameter().getType() == SoftAssertions.class) {
+			return getStore(extensionContext).getOrComputeIfAbsent(SoftAssertions.class);
+		}
+		return getStore(extensionContext).getOrComputeIfAbsent(BDDSoftAssertions.class);
```

Comment:
Could we do this ?
```java
return getStore(extensionContext).getOrComputeIfAbsent(parameterContext.getParameter().getType());
```

Reply:
Yes, I suppose we could do that. It looks a bit like voodoo, but with a comment I think it would fine. 😉

Reply:
Done in latest commit: 21f5677